core/src/main/java/org/bitcoinj/evolution/SimplifiedMasternodeList.java

```diff
@@ -92,15 +97,10 @@ protected void parse() throws ProtocolException {
             mnMap.put(hash, mn);
         }
 
+        // read the number of properties, which should be zero
         size = (int)readVarInt();
-        mnUniquePropertyMap = new HashMap<Sha256Hash, Pair<Sha256Hash, Integer>>(size);
-        for(long i = 0; i < size; ++i)
-        {
-            Sha256Hash hash = readHash();
-            Sha256Hash first = readHash();
-            int second = (int)readUint32();
-            mnUniquePropertyMap.put(hash, new Pair<Sha256Hash, Integer>(first, second));
-        }
+        Preconditions.checkArgument(size == 0, "There is an offset error with this data file, rejecting...");
```

Comment:
This will solve the Out Of Memory exception when loading the mnlist file from storage.  We don't use the mnUniquePropertyMap and save a zero size, but in some cases the mnlist file is corrupted and the size turns out to be non zero and perhaps rather large.  That size is used to create a new HashMap.

Now we will enforce that the size is zero and then fail if it is.  This will result in the mnlist file being ignored and recreated.